ompi/mca/pml/ucx/pml_ucx.c

```diff
@@ -414,31 +415,37 @@ int mca_pml_ucx_del_procs(struct ompi_proc_t **procs, size_t nprocs)
             continue;
         }
 
+        proc->proc_endpoints[OMPI_PROC_ENDPOINT_TAG_PML] = NULL;
+
         PML_UCX_VERBOSE(2, "disconnecting from rank %d", proc->super.proc_name.vpid);
         dreq = ucp_disconnect_nb(ep);
         if (dreq != NULL) {
             if (UCS_PTR_IS_ERR(dreq)) {
                 PML_UCX_ERROR("ucp_disconnect_nb(%d) failed: %s",
                               proc->super.proc_name.vpid,
                               ucs_status_string(UCS_PTR_STATUS(dreq)));
+                continue;
             } else {
                 dreqs[num_reqs++] = dreq;
             }
         }
 
-        proc->proc_endpoints[OMPI_PROC_ENDPOINT_TAG_PML] = NULL;
-
-        if ((int)num_reqs >= ompi_pml_ucx.num_disconnect) {
+        if (num_reqs && ((int)num_reqs >= ompi_pml_ucx.num_disconnect)) {
             mca_pml_ucx_waitall(dreqs, &num_reqs);
         }
     }
-
-    mca_pml_ucx_waitall(dreqs, &num_reqs);
+    if (num_reqs) {
```

Comment:
why needed? mca_pml_ucx_waitall() should handle num_reqs==0

Reply:
what issue? maybe need to convert ` mca_pml_ucx_waitall(void **reqs, int *count_p)` to ` mca_pml_ucx_waitall(void **reqs, int count)`?

Reply:
@yosefe I just suppressed this warning